controllers/datamovement_controller.go

```diff
@@ -192,62 +218,175 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
+	dm.Status.CommandStatus = &nnfv1alpha1.NnfDataMovementCommandStatus{}
+	cmdStatus := nnfv1alpha1.NnfDataMovementCommandStatus{}
+	cmdStatus.Command = cmd.String()
+	log.Info("Running Command", "cmd", cmdStatus.Command)
 
 	if err := r.Status().Update(ctx, dm); err != nil {
 		return ctrl.Result{}, err
 	}
 
-	// Expand the context with cancel and store it in the map so the cancel
-	// function can be used in another reconciler loop. Also add NamespacedName
-	// so we can retrieve the resource.
-	ctxCancel, cancel := context.WithCancel(ctx)
-	r.contexts.Store(dm.Name, dataMovementCancelContext{
-		ctx:    ctxCancel,
-		cancel: cancel,
-	})
-
 	// Execute the go routine to perform the data movement
 	go func() {
-		var cmd *exec.Cmd
+		// Use a MultiWriter so that we can parse the output and save the full output at the end
+		var combinedOutBuf, parseBuf bytes.Buffer
+		cmd.Stdout = io.MultiWriter(os.Stdout, &parseBuf, &combinedOutBuf)
+		cmd.Stderr = cmd.Stdout // Combine stderr/stdout
+
+		// Use channels to sync progress collection and cmd.Wait().
+		chCommandDone := make(chan bool, 1)
+		chProgressDone := make(chan bool)
+
+		// Start the data movement command
+		cmd.Start()
+
+		// While the command is running, capture and process the output. Read lines until EOF to
+		// ensure we have the latest output. Then use the last regex match to obtain the most recent
+		// progress.
+		go func() {
+			// If progress collection interval is >= 1s, otherwise skip this if
+			if !progressCollectionEnabled(progressCollectInterval) {
+				log.Info("Skipping progress collection - collection interval is less than 1s", "collectInterval", progressCollectInterval)
+				return
+			}
 
-		// TODO: UserId/GroupId
+			var progressInt32 *int32
+			var elapsed metav1.Duration
+			elapsed.Duration = 0
+			progressStart := metav1.NowMicro()
+
+			// Perform the actual collection and update logic
+			parseAndUpdateProgress := func() {
+				progressOutput := ""
+
+				// Read all lines of output until EOF
+				for {
+					line, err := parseBuf.ReadString('\n')
+					if err == io.EOF {
+						break
+					} else if err != nil {
+						log.Error(err, "failed to read progress output")
+					}
+					progressOutput += line
+				}
+
+				// Get the times now before we start the parsing
+				progressNow := metav1.NowMicro()
+				elapsed.Duration = progressNow.Time.Sub(progressStart.Time)
+
+				// Find all matching progress matches but we only care about the
+				// last (most recent one)
+				groups := progressRe.FindAllStringSubmatch(progressOutput, -1)
+				if len(groups) > 0 {
+					mostRecent := groups[len(groups)-1]
+
+					// Get the progress and convert to int
+					progress, err := strconv.Atoi(mostRecent[1])
+					if err == nil {
+						// Initialize the pointer if we haven't done so already.
+						// We don't want to do this above since we do not want an
+						// initial value of 0 unless we actually have a progress
+						if progressInt32 == nil {
```

Comment:
Did you investigate if this is necessary? I thought Go was smart enough that this could just be `cmdStatus.ProgressPercentage = &progress`

Reply:
Turns out I can just do this:
```
progressInt32 := int32(progress)
cmdStatus.ProgressPercentage = &progressInt32
```